### ISSUE TITLE
Can't check if iap has_purchased right after iap.register

### ISSUE THREAD

**Issue by Istreb** [renpy/renpy#3322]
Hey everyone!
while running the following code in options.rpy:
```
iap.register("cl_dlc", identifier="cl_dlc")
if renpy.variant("pc") or iap.has_purchased("cl_dlc"):
  ...
```
The app crashes and I get the error in logcat:

```
2022-03-08 19:17:41.402 2028-5166/? I/python:   File "game/script/paid/vars.rpy", line 3, in <module>
2022-03-08 19:17:41.402 2028-5166/? I/python:   File "renpy/common/00iap.rpy", line 492, in has_purchased
2022-03-08 19:17:41.402 2028-5166/? I/python: AttributeError: 'NoneType' object has no attribute 'get'
```

vars.rpy, line 3 - is the last line in the example above and it's iap.has_purchased("cl_dlc")

My only guess is that iap.register takes some time to finish and while executing iap.has_purchased, register hasn't finished yet. So we have kind of "race condition" here

But I'm not sure and even if it's true - don't know how to fix it.

**Comment by renpytom** [comment]
The IAP initializes at the end of the init phase.

**Comment by renpytom** [comment]
445dbe9f4 adds support for initializing IAP in the middle of the init phase. While it won't be released until 7.5/8.0 is out, you should be able to fairly easily backport this change.
